plugins/modules/swim_intent.py

```diff
@@ -1362,6 +1379,125 @@ def get_diff_merged(self, config):
+
+            response = self.dnac._exec(
+                family="software_image_management_swim",
+                function='get_golden_tag_status_of_an_image',
+                op_modifies=True,
+                params=image_params
+            )
+            self.log("Received API response from 'get_golden_tag_status_of_an_image': {0}".format(str(response)), "DEBUG")
+
+            response = response.get('response')
+            if response:
+                image_status = response['taggedGolden']
+                if image_status == tag_image_golden:
+                    if tag_image_golden:
+                        self.msg = """The requested image '{0}' has been tagged as golden in the Cisco Catalyst Center and image status
+                                has been verified.""".format(name)
+                        self.log(self.msg, "INFO")
+                    else:
+                        self.msg = """The requested image '{0}' has been un-tagged as golden in the Cisco Catalyst Center and
+                                image status has been verified.""".format(name)
+                        self.log(self.msg, "INFO")
+            else:
+                self.log("""Mismatch between the playbook input for tagging/un-tagging image as golden and the Cisco Catalyst Center indicates that
+                         the tagging/un-tagging task was not executed successfully.""", "INFO")
+
+        if self.want.get("distribution_details"):
+            image_id = self.have.get("distribution_image_id")
+
+            if self.have.get("distribution_device_id"):
+                if self.single_device_distribution:
+                    self.msg = """The requested image with device id '{0}' has been distributed successfully in the Cisco Catalyst Center and
+                             image status has been verified.""".format(self.have.get("distribution_device_id"))
+                    self.log(self.msg, "INFO")
+                else:
+                    self.log("""Mismatch between the playbook input for distributing image to device with id '{0}' and the Cisco Catalyst Center indicates that
+                         the distribution task was not executed successfully.""".format(self.have.get("distribution_device_id")), "INFO")
+            elif self.complete_successful_distribution:
+                self.msg = """The requested image with id '{0}' distributed successfully on all devices in that particular site
+                         in the Cisco Catalyst Center.""".format(image_id)
+                self.log(self.msg, "INFO")
+            elif self.partial_successful_distribution:
+                self.msg = "The requested image with id '{0}' distributed partially on some devices in the Cisco Catalyst Center.".format(image_id)
+                self.log(self.msg, "INFO")
+            else:
+                self.msg = "The requested image with id '{0}' failed to distributed on devices in the Cisco Catalyst Center.".format(image_id)
```

Comment:
The requested image, with ID '{0}', failed to be distributed across devices in the Cisco Catalyst Center.".format(image_id)

Also which devices???

Reply:
For now I added the message, and we are printing device ips while executing the process. But here it'll be little complicated so will take once all pending items get done.